### PR TITLE
feat(@esri/arcgis-rest-items): add support for item Info upload/removal

### PR DESCRIPTION
**Pull request by dbouwman**
Added functions `updateItemInfo` and `removeItemInfo` to `@esri/arcgis-rest-items` to allow us to work with the `/info` files? resources? Needed to support Survey construction in Hub.

Also adds a yarn.lock to the Vue demo... 

AFFECTS PACKAGES:
@esri/arcgis-rest-items
@esri/arcgis-rest-demo-vue-with-popup

### CODE REVIEW

File: packages/arcgis-rest-items/src/helpers.ts [hunk 1]
```diff
@@ -49,6 +49,35 @@ export interface IItemResourceRequestOptions extends IItemIdRequestOptions {
...
+
+export interface IItemRemoveInfoRequestOptions extends IItemIdRequestOptions {
+  /**
+   * The file to remove
+   */
+  fileName: string;
+  /**
+   * If provided, adds the info in a sub-folder
+   */
+  folderName?: boolean;
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/helpers.ts]: doesn't look like you're actually using `folderName`...

File: packages/arcgis-rest-items/src/remove.ts [hunk 2]
```diff
@@ -56,6 +57,38 @@ export function removeItemResource(
...
+ *   .then(response) // --> {success:true|false}
+ * ```
+ * Remove an info file associated with an item. See the [REST Documentation](https://developers.arcgis.com/rest/users-groups-and-items/delete-info.htm) for more information.
+ *
+ * @param requestOptions - Options for the request
+ * @returns A Promise that deletes an item resource.
+ */
+export function removeItemInfo(
+  requestOptions: IItemRemoveInfoRequestOptions
+): Promise<any> {
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/remove.ts]:
```ts
Promise<{ "success": boolean >
```
no need to create an interface to document the response structure.

File: packages/arcgis-rest-items/src/remove.ts [hunk 3]
```diff
@@ -56,6 +57,38 @@ export function removeItemResource(
...
+ *   id: '3ef',
+ *   infoFile: 'bigkahuna.json',
+ *   authentication
+ * })
+ *   .then(response) // --> {success:true|false}
+ * ```
+ * Remove an info file associated with an item. See the [REST Documentation](https://developers.arcgis.com/rest/users-groups-and-items/delete-info.htm) for more information.
+ *
+ * @param requestOptions - Options for the request
+ * @returns A Promise that deletes an item resource.
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/remove.ts]: we need to differentiate from _regular_ item [resources](https://developers.arcgis.com/rest/users-groups-and-items/add-resources.htm).
```js
@returns A Promise that deletes an item info file.
```

File: packages/arcgis-rest-items/test/mocks/resources.ts [hunk 4]
```diff
@@ -29,3 +29,10 @@ export const UpdateItemResourceResponse: any = {
   owner: "jsmith",
   folder: null
 };
+
+export const UpdateItemInfoResponse: any = {
```
Comment (jgravois) [inline packages/arcgis-rest-items/test/mocks/resources.ts]:
```ts
export const UpdateItemInfoResponse: IItemResourceResponse = {
```

File: packages/arcgis-rest-items/src/update.ts [hunk 5]
```diff
@@ -109,6 +110,68 @@ export function updateItemResource(
...
+ *
+ * Valid File types: JSON, XML, CFG, TXT, PBF, and PNG
+ * Max File Size: 100K
+ *
+ * @param requestOptions  Options for the request
+ * @returns A Promise that add/updates an item info.
+ */
+export function updateItemInfo(
+  requestOptions: IItemUpdateInfoRequestOptions
+): Promise<any> {
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/update.ts]: Promise<IItemResourceResponse>

luckily we already have an interface defined for this.

https://github.com/Esri/arcgis-rest-js/blob/8be88a399a35e1dc31f77f377fb095489e055b2f/packages/arcgis-rest-items/src/helpers.ts#L79-L84

File: packages/arcgis-rest-items/src/update.ts [hunk 6]
```diff
@@ -109,6 +110,68 @@ export function updateItemResource(
...
+      if (ext === "xml") {
+        type = "xml";
+      }
+      requestOptions.params.file = new Blob([requestOptions.content], {
+        type: `text/${type}`
+      });
+    }
+  }
+
+  return request(url, requestOptions);
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/update.ts]: unless i'm missing something, it doesn't look like you're actually doing anything with `folderName` in this method either.

**Comment by dbouwman** [comment]
Thanks @jgravois I'll clean this up in the next few days, and try to suss out how to send the folder. I'll also puzzle out the CI failure... she's all happy locally (of course)

File: packages/arcgis-rest-items/src/remove.ts [hunk 7]
```diff
@@ -56,6 +57,38 @@ export function removeItemResource(
   return request(url, requestOptions);
 }
 
+/**
+ * ```js
+ * import { removeItemInfo } from '@esri/arcgis-rest-items';
+ * //
+ * removeItemInfo({
+ *   id: '3ef',
+ *   infoFile: 'bigkahuna.json',
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/remove.ts]: this should be:
```js
fileName: 'bigkahuna.json',
```

**Comment by jgravois** [comment]
closing for now, but happy to help get this across the 🏁 line anytime you come back around to it.

**Pull request closed without merging**